odoo/addons/test_read_group/tests/test_private_read_group.py

```diff
@@ -686,3 +686,13 @@ def test_groupby_many2many(self):
                 (User, ["Donkey Kong"]),                            # tasks of nobody
             ],
         )
+
+    def test_float_aggregate(self):
+        records = self.env['test_read_group.aggregate'].create({'numeric_value': 42.42})
+        self.assertEqual(records._read_group(
+                [('id', 'in', records.ids)],
+                [],
+                ['numeric_value:array_agg'],
+            ),
+            [([42.42],)],
+        )
```

Comment:
This does not test the returned type.  Try for instance `Decimal("2.5") == 2.5`, it's `True`.
```suggestion
        [[result]] = records._read_group([('id', 'in', records.ids)], [], ['numeric_value:array_agg'])
        self.assertIsInstance(result, list)
        self.assertIsInstance(result[0], float)
```

Reply:
Whut ? I tested it.

Weird stuff: 
![image](https://github.com/odoo/odoo/assets/53175387/e92961c1-2d62-4c7c-a135-9349ef49a4c4)